onnxruntime/python/onnxruntime_pybind_state.cc

```diff
@@ -352,151 +352,157 @@ std::unique_ptr<IExecutionProvider> CreateExecutionProviderInstance(
+              }
+            } else if (option.first == "trt_engine_cache_path") {
+              if (!option.second.empty()) {
+                cache_path = option.second;
+                params.trt_engine_cache_path = cache_path.c_str();
+              } else {
+                ORT_THROW("[ERROR] [TensorRT] The value for the key 'trt_engine_cache_path' should be a path string i.e. 'engine_cache'.\n");
+              }
+            } else if (option.first == "trt_engine_decryption_enable") {
+              if (option.second == "True" || option.second == "true") {
+                params.trt_engine_decryption_enable = true;
+              } else if (option.second == "False" || option.second == "false") {
+                params.trt_engine_decryption_enable = false;
+              } else {
+                ORT_THROW("[ERROR] [TensorRT] The value for the key 'trt_engine_decryption_enable' should be a boolean i.e. 'True' or 'False'. Default value is False.\n");
+              }
+            } else if (option.first == "trt_engine_decryption_lib_path") {
+              if (!option.second.empty()) {
+                lib_path = option.second;
+                params.trt_engine_decryption_lib_path = lib_path.c_str();
+              } else {
+                ORT_THROW("[ERROR] [TensorRT] The value for the key 'trt_engine_decryption_lib_path' should be a path string i.e. 'decryption_lib'.\n");
+              }
+            } else if (option.first == "trt_force_sequential_engine_build") {
+              if (option.second == "True" || option.second == "true") {
+                params.trt_force_sequential_engine_build = true;
+              } else if (option.second == "False" || option.second == "false") {
+                params.trt_force_sequential_engine_build = false;
+              } else {
+                ORT_THROW("[ERROR] [TensorRT] The value for the key 'trt_force_sequential_engine_build' should be a boolean i.e. 'True' or 'False'. Default value is False.\n");
+              }
+            } else {
+              ORT_THROW("Invalid TensorRT EP option: ", option.first);
+            }
           }
+          return onnxruntime::CreateExecutionProviderFactory_Tensorrt(&params)->CreateProvider();
         } else {
-          ORT_THROW("Invalid TensorRT EP option: ", option.first);
+          return onnxruntime::CreateExecutionProviderFactory_Tensorrt(cuda_device_id)->CreateProvider();
         }
-      }
-      return onnxruntime::CreateExecutionProviderFactory_Tensorrt(&params)->CreateProvider();
     } else {
-      return onnxruntime::CreateExecutionProviderFactory_Tensorrt(cuda_device_id)->CreateProvider();
+      ORT_THROW("Please install the correct version TensorRT as well as CUDA and cuDNN as mentioned in the TensorRT requirements page (https://onnxruntime.ai/docs/execution-providers/TensorRT-ExecutionProvider.html#requirements), make sure they're in the PATH, and that your GPU is supported.");
```

Comment:
We do not want to unconditionally throw here.
this else condition can be hit on a system without cuda/tensorrt installed.
I think we should follow the cuda example and check for whether CUDA_PATH is defined as a hint that this is a system with nvidia gpu/cuda installed.

Reply:
can you please test the package on a system with just CUDA , TRT + CUDA and one without CUDA to confirm all those scenarios work properly. thanks!

Reply:
Rewrote the condition as you suggested to throw exception.
Will test the package once it's been built by CI Pipelines.